data/spells/scripts/support/levitate.lua

```diff
@@ -0,0 +1,40 @@
+function onCastSpell(creature, variant)
+	local currentPos = creature:getPosition()
+	local destPos = creature:getPosition()
+	destPos:getNextPosition(creature:getDirection())
+
+	local ret = RETURNVALUE_NOTPOSSIBLE
+	local param = variant:getString():lower()
+	if param == "up" then
+		if currentPos.z ~= 8 then
+			local tile = Tile(currentPos.x, currentPos.y, currentPos.z - 1)
+			if tile == nil or (tile:getGround() == nil and not tile:hasFlag(TILESTATE_IMMOVABLEBLOCKSOLID)) then
```

Comment:
Not sure if the check should be like what @WibbenZ told. I saw a few scripts in the datapack that use == nil.